src/darwin/Framework/CHIP/MTROTAHeaderParser.mm

```diff
@@ -0,0 +1,72 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTROTAHeaderParser.h"
+
+#import "MTRError.h"
+#import "MTRError_Internal.h"
+
+#include <lib/core/OTAImageHeader.h>
+
+@implementation MTROTAHeaderParser
++ (nullable NSDictionary *)dictFromData:(NSData *)data error:(NSError * __autoreleasing *)error
+{
+    chip::OTAImageHeaderParser parser;
+
+    parser.Init();
+
+    if (!parser.IsInitialized()) {
+        *error = [NSError errorWithDomain:MTRErrorDomain code:MTRErrorCodeGeneralError userInfo:nil];
+        return nil;
+    }
+
+    chip::ByteSpan buffer(static_cast<const unsigned char *>([data bytes]), [data length]);
+    chip::OTAImageHeader header;
+
+    CHIP_ERROR err = parser.AccumulateAndDecode(buffer, header);
+    if (err != CHIP_NO_ERROR) {
+        *error = [MTRError errorForCHIPErrorCode:err];
+        parser.Clear();
+        return nil;
+    }
+
+    auto headerDict = [[NSMutableDictionary alloc] init];
+    headerDict[@"vendorId"] = @(header.mVendorId);
```

Comment:
Spoke to Vivien off-line to make these keys `static const NSString *`

Reply:
I end up using a custom interface. That should fit your needs here:
```
@interface MTROTAHeader : NSObject

@property (nonatomic, strong) NSNumber * vendorID;
@property (nonatomic, strong) NSNumber * productID;
@property (nonatomic, strong) NSNumber * payloadSize;
@property (nonatomic, strong) NSNumber * softwareVersion;
@property (nonatomic, strong) NSString * softwareVersionString;
@property (nonatomic, strong) NSString * releaseNotesURL;
@property (nonatomic, strong) NSData * imageDigest;
@property (nonatomic, assign) MTROTAImageDigestType imageDigestType;
@property (nonatomic, strong) NSNumber * minApplicableVersion;
@property (nonatomic, strong) NSNumber * maxApplicableVersion;

@end
```